src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -107,14 +110,17 @@ class FileService(fileRepository: FileRepository,
     val currentCursor = input.currentCursor.map(UUID.fromString)
     val limit = input.limit
     val maxFiles: Int = min(limit, filePageMaxLimit)
-
     for {
       response: Seq[FileRow] <- fileRepository.getPaginatedFiles(consignmentId, maxFiles, currentCursor, filters)
       fileIds = Some(response.map(_.fileid).toSet)
-      fileMetadata <- fileMetadataService.getFileMetadata(consignmentId, fileIds)
-      ffidMetadataList <- ffidMetadataService.getFFIDMetadata(consignmentId, fileIds)
-      avList <- avMetadataService.getAntivirusMetadata(consignmentId, fileIds)
-      ffidStatus <- fileStatusService.getFileStatus(consignmentId, fileIds)
+      getFileMetadata = fileMetadataService.getFileMetadata(consignmentId, fileIds)
+      getFfidMetadataList = ffidMetadataService.getFFIDMetadata(consignmentId, fileIds)
+      getAvList = avMetadataService.getAntivirusMetadata(consignmentId, fileIds)
+      getFfidStatus = fileStatusService.getFileStatus(consignmentId, fileIds)
```

Comment:
Similar to the other comment.

Reply:
Same comment as the one I made above; I think we need to ensure that `fileAndMetadataList` is completed first.